lib/sidekiq/client.rb

```diff
@@ -67,15 +67,19 @@ def initialize(redis_pool = nil)
     #   push('queue' => 'my_queue', 'class' => MyWorker, 'args' => ['foo', 1, :bat => 'bar'])
     #
     def push(item)
-      normed = normalize_item(item)
-      payload = process_single(item["class"], normed)
+      payload = payload_for_push(item)
 
       if payload
         raw_push([payload])
         payload["jid"]
       end
     end
 
+    def payload_for_push(item)
```

Comment:
This creates a new public API. Can't do that.

Reply:
One option I can think of is to move this method to `worker.rb` but that may not be maintainable as logic for private method `normalize_item` would be in 2 places.
Any suggestions on how to proceed?